### PR TITLE
HBASE-25122 [Flake Test][branch-1] TestExportSnapshotWithTemporaryDirectory

### PR DESCRIPTION
**Pull request by Reidddddd**
[HBASE-25122](https://issues.apache.org/jira/browse/HBASE-25122)

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m  6s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 30s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 53s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 54s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   2m 11s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 52s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m 56s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 52s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 54s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 54s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 56s |  hbase-server: The patch generated 0 new + 3 unchanged - 3 fixed = 3 total (was 6)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 41s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 40s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 52s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   4m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 149m 43s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 42s |  The patch does not generate ASF License warnings.  |
|  |   | 207m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2472 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 86ce5299aec8 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2472/out/precommit/personality/provided.sh |
| git revision | branch-1 / e719a5b |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/1/testReport/ |
| Max. process+thread count | 3875 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshotWithTemporaryDirectory.java [hunk 1]
```diff
@@ -31,24 +27,21 @@
...
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
-    setUpBaseConf(TEST_UTIL.getConfiguration());
+    Configuration conf = TEST_UTIL.getConfiguration();
+    TestExportSnapshot.setUpBaseConf(conf);
     TEST_UTIL.startMiniCluster(3);
+    Path rootDir = TEST_UTIL.getMiniHBaseCluster().getMaster().getMasterFileSystem().getRootDir();
+    LOG.info("Root dir: " + rootDir);
+    conf.set(SnapshotDescriptionUtils.SNAPSHOT_WORKING_DIR,
+      new Path(rootDir.getParent(), ".tmpdir").toUri().toString());
```
Comment (virajjasani) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshotWithTemporaryDirectory.java:38]: Agree, this is reasonable.

**Review by virajjasani — APPROVED** [review]
+1

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 20s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 54s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  compile  |   0m 53s |  branch-1 passed with JDK v1.7.0_272  |
| +1 :green_heart: |  checkstyle  |   2m  7s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 45s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  0s |  branch-1 passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 52s |  branch-1 passed with JDK v1.7.0_272  |
| +0 :ok: |  spotbugs  |   3m 50s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 46s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 57s |  hbase-server: The patch generated 0 new + 3 unchanged - 3 fixed = 3 total (was 6)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 54s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed with JDK v1.8.0_262  |
| +1 :green_heart: |  javadoc  |   0m 52s |  the patch passed with JDK v1.7.0_272  |
| +1 :green_heart: |  findbugs  |   4m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 165m 45s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  The patch does not generate ASF License warnings.  |
|  |   | 217m 12s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestReplicasClient |
|   | hadoop.hbase.master.balancer.TestStochasticLoadBalancer2 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.13 Server=19.03.13 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2472 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 55d15ca664bb 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2472/out/precommit/personality/provided.sh |
| git revision | branch-1 / e76e7c9 |
| Default Java | 1.7.0_272 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_262 /usr/lib/jvm/zulu-7-amd64:1.7.0_272 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/2/testReport/ |
| Max. process+thread count | 3645 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2472/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**